src/Symfony/Component/HtmlSanitizer/Parser/NativeParser.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer\Parser;
+
+/**
+ * Parser using PHP 8.4's new Dom API.
+ */
+final class NativeParser
+{
+    public function __construct()
+    {
+        if (\PHP_VERSION_ID < 80400) {
+            throw new \LogicException(self::class.' requires PHP 8.4 or higher.');
+        }
+    }
+
+    public function parse(string $html, string $context = 'body'): ?\Dom\Node
+    {
+        $document = @\Dom\HTMLDocument::createFromString(\sprintf('<!DOCTYPE html><%s>%s</%1$s>', $context, $html), \Dom\HTML_NO_DEFAULT_NS);
+        $fragment = $document->createDocumentFragment();
+
+        $context = $document->getElementsByTagName($context)->item(0);
+        $count1 = $context->childElementCount;
+
+        $fragment = $document->createDocumentFragment();
+        while ($context->firstChild) {
+            $fragment->appendChild($context->firstChild);
+        }
+
+        $context->innerHTML = $html;
```

Comment:
Here we parse a 2nd time with a different algo and we take the results that yields the most nodes.
Without this,`'<tbody>Lorem ipsum</tbody>'` is turned into `'Lorem ipsum'`, which differs from how MastermindsParser works.

Reply:
2nd time parsing removed by passing the context tag element to the parser.
This means a `<td>` won't be accepted anymore in a `<body>`. A `<table>` context must be specified instead.